### PR TITLE
Move ring buffer out of toxcore/util into toxav.

### PR DESCRIPTION
**Pull request by iphydf**
Toxcore itself doesn't use this data structure. Only toxav does, so now
toxav owns the code for it.

### CODE REVIEW

File: testing/av_test.c [hunk 1]
```diff
@@ -22,6 +22,20 @@
  *   -lopencv_highgui -lopencv_imgproc -lsndfile -pthread -lvpx -lopus -lsodium -lportaudio
  */
 
+// XXX: Hack because toxav doesn't really expose ring_buffer, but this av test
+// uses it.
+#define RingBuffer TestRingBuffer
+#define rb_full  test_rb_full
```
Comment (GrayHatter) [inline testing/av_test.c]: Most of these aren't actually used in this test. They should be removed to make the code easier to read

File: toxav/ring_buffer.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#ifndef RING_BUFFER_H
```
Comment (GrayHatter) [inline toxav/ring_buffer.h]: This header needs a comment as to the point of RingBuffer (and maybe a rename!), 

FYI ring buffer stores AV packets received so ToxAV can calculate the mean on how many packets are lost/received.

Comment (iphydf) [inline toxav/ring_buffer.h]: I don't know the point of RingBuffer. This PR just moves it out of toxcore where it doesn't belong. @mannol can add documentation.

File: testing/av_test.c (hunk 1 above)
Comment (iphydf) [inline testing/av_test.c]: They aren't used, but if linking statically, not renaming them will cause multiple definition errors. We need to rename all of them.

Comment (nurupo) [inline testing/av_test.c]: Maybe mention the static linking issue in the comment too.

File: toxav/ring_buffer.h (hunk 2 above)
Comment (mannol) [inline toxav/ring_buffer.h]: @GrayHatter It's literally a name of the data type. I don't see a point in renaming it for this special case.

File: testing/av_test.c (hunk 1 above)
Comment (mannol) [inline testing/av_test.c]: What's the need for ring_buffer compilation here?

Comment (iphydf) [inline testing/av_test.c]: @nurupo done.

**Comment by iphydf** [comment]
Since I can't reply to the review thread (@mannol do you still like github reviews better than reviewable?):
- What's the need for ring buffer: I don't know, it's used for something, so I'm making it work.

**Comment by mannol** [comment]
I do. We should just avoid force pushing; at least until the review is complete.
I'm not asking about the need for ring buffer, I'm asking why isn't ```#include "ring_buffer.h"``` sufficient? Symbols should be defined in ```libtoxav.*```.

**Comment by iphydf** [comment]
The symbols are not part of the ABI, and since #117, toxav does not export those symbols in the .so file anymore (on linux).

File: toxav/ring_buffer.h (hunk 2 above)
Comment (GrayHatter) [inline toxav/ring_buffer.h]: OH... I've been reading it as ring, as in ringing call... too much AV on the brain

**Pull request merged**